File: Java/Migration/unit-tests-csharp/unit-tests-csharp/P01CoreRunner/Ex02CategoriesGroups.cs
using NUnit.Framework;

namespace unit_tests_csharp.P01CoreRunner
{
  public class Ex02CategoriesGroups
  {
    //TODO show how resharper groups this test
    [Test, Category("SlowTest"), Category("NotReallyAUnitTests")]
    [Category("JustFoolingAround")]
    public void ShouldDoWhatever()
    {
      //GIVEN

      //WHEN

      //THEN
    }
  }
}

File: Java/Migration/unit-tests-java/src/test/java/p01_core_runner/Ex02CategoriesGroups.java
package p01_core_runner;

import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class Ex02CategoriesGroups {

  @Test(groups = {
      "SlowTest",
      "NotReallyAUnitTests",
      "JustFoolingAround"})
  public void shouldDoWhatever() {
    //GIVEN

    //WHEN

    //THEN
    assertEquals(2,2);
  }

  @Test(groups = {""}) //TODO show how known groups are suggested
  public void shouldDoWhatever2() {
    //GIVEN

    //WHEN

    //THEN
    assertEquals(2,2);
  }

}
